crates/op-rbuilder/src/builders/context.rs

```diff
@@ -600,6 +603,66 @@ impl<ExtraCtx: Debug + Default> OpPayloadBuilderCtx<ExtraCtx> {
             // append sender and transaction to the respective lists
             info.executed_senders.push(tx.signer());
             info.executed_transactions.push(tx.into_inner());
+
+            if is_success && let Some(backrun_bundles) = self.backrun_bundle_store.get(&tx_hash) {
+                self.metrics.backrun_target_txs_found_total.increment(1);
+
+                for stored_bundle in backrun_bundles {
+                    info!(
+                        target: "payload_builder",
+                        message = "Executing backrun bundles",
+                        tx_hash = ?tx_hash,
+                        bundle_id = ?stored_bundle.bundle_id,
+                    );
+
+                    for backrun_tx in stored_bundle.backrun_txs {
+                        let ResultAndState { result, state } = match evm.transact(&backrun_tx) {
+                            Ok(res) => res,
+                            Err(err) => {
+                                return Err(PayloadBuilderError::evm(err));
+                            }
+                        };
+
+                        let backrun_gas_used = result.gas_used();
+                        let is_backrun_success = result.is_success();
+
+                        if !is_backrun_success {
```

Comment:
The logic for "not committing backrun tx if it fails" breaks in case you have multiple backrun_txs in the bundle. 

For example, if there are 2 backrun_txs and you commit 1 successfully and then 2 unsuccessfully then 1 will not be reverted. 

Is this intended?